File: base/src/main/java/com/zhangteng/base/utils/BeanUtils.java
package com.zhangteng.base.utils;

import java.lang.reflect.Field;
import java.lang.reflect.Modifier;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class BeanUtils {
    /**
     * Object转Map
     */
    public static Map<String, Object> getObjectToMap(Object obj) {
        Map<String, Object> map = new HashMap<>();
        Class<?> clazz = obj.getClass();
        System.out.println(clazz);
        for (Field field : clazz.getDeclaredFields()) {
            field.setAccessible(true);
            try {
                String fieldName = field.getName();
                Object value = field.get(obj);
                if (value == null) {
                    value = "";
                }
                map.put(fieldName, value);
            } catch (Exception ignore) {
            }
        }
        return map;
    }

    /**
     * Map转Object
     */
    public static Object mapToObject(Map map, Class<?> beanClass) {
        try {
            if (map == null)
                return null;
            Object obj = beanClass.newInstance();
            Field[] fields = getAllFields(obj);
            for (Field field : fields) {
                int mod = field.getModifiers();
                if (Modifier.isStatic(mod) || Modifier.isFinal(mod)) {
                    continue;
                }
                field.setAccessible(true);
                if (map.containsKey(field.getName())) {
                    field.set(obj, map.get(field.getName()));
                }
            }

            return obj;
        } catch (Exception ignore) {
            return null;
        }
    }

    private static Field[] getAllFields(Object object) {
        Class clazz = object.getClass();
        List<Field> fieldList = new ArrayList<>();
        while (clazz != null) {
            fieldList.addAll(new ArrayList<>(Arrays.asList(clazz.getDeclaredFields())));
            clazz = clazz.getSuperclass();
        }
        Field[] fields = new Field[fieldList.size()];
        fieldList.toArray(fields);
        return fields;
    }
}

File: base/src/main/java/com/zhangteng/base/utils/BeanUtils.kt
package com.zhangteng.base.utils

import java.lang.reflect.Field
import java.lang.reflect.Modifier
import java.util.*

object BeanUtils {
    /**
     * Object转Map
     */
    fun getObjectToMap(obj: Any?): MutableMap<String?, Any?>? {
        if (obj == null) return null
        val map: MutableMap<String?, Any?> = HashMap()
        val clazz: Class<*> = obj.javaClass
        println(clazz)
        for (field in clazz.declaredFields) {
            field.isAccessible = true
            try {
                val fieldName = field.name
                var value = field[obj]
                if (value == null) {
                    value = ""
                }
                map[fieldName] = value
            } catch (ignore: Exception) {
            }
        }
        return map
    }

    /**
     * Map转Object
     */
    fun mapToObject(map: MutableMap<*, *>?, beanClass: Class<*>?): Any? {
        return try {
            if (map == null || beanClass == null) return null
            val obj = beanClass.newInstance()
            val fields = getAllFields(obj)
            for (field in fields!!) {
                val mod = field!!.modifiers
                if (Modifier.isStatic(mod) || Modifier.isFinal(mod)) {
                    continue
                }
                field.isAccessible = true
                if (map.containsKey(field.name)) {
                    field.set(obj, map[field.name])
                }
            }
            obj
        } catch (ignore: Exception) {
            null
        }
    }

    private fun getAllFields(`object`: Any?): MutableList<Field?>? {
        if (`object` == null) return null
        var clazz: Class<*>? = `object`.javaClass
        val fieldList: MutableList<Field?> = ArrayList()
        while (clazz != null) {
            fieldList.addAll(ArrayList(listOf(*clazz.declaredFields)))
            clazz = clazz.superclass
        }
        return fieldList
    }
}